dashboard/test/controllers/script_levels_controller_test.rb

```diff
@@ -712,4 +712,33 @@ class ScriptLevelsControllerTest < ActionController::TestCase
     assert_response :success
   end
 
+  def create_admin_script
+    create(:script, admin_required: true).tap do |script|
+      create :script_level, script: script
+    end
+  end
+
+  test "should not get show of admin script if not signed in" do
+    admin_script = create_admin_script
+
+    get :show, script_id: admin_script.name, stage_id: 1, id: 1
+    assert_redirected_to_sign_in
+  end
+
+  test "should not get show of admin script if signed in as not admin" do
+    admin_script = create_admin_script
+
+    sign_in @not_admin
+    get :show, script_id: admin_script.name, stage_id: 1, id: 1
+    assert_response :forbidden
+  end
+
+  test "should get show of admin script if signed in as admin" do
+    admin_script = create_admin_script
+
+    sign_in @admin
+    get :show, script_id: admin_script.name, stage_id: 1, id: 1
+    assert_response :success
+  end
```

Comment:
not as written because I need to create a script. I could maybe change the generate_admin_only_tests_for to yield to a block instead of take params so I could have both the get and the create in there